crates/ruff/src/linter.rs

```diff
@@ -526,3 +488,68 @@ This indicates a bug in `{}`. If you could open an issue at:
         });
     }
 }
+
+#[allow(clippy::print_stderr)]
+fn report_failed_to_converge_error(path: &Path, transformed: &str) {
+    #[cfg(debug_assertions)]
+    {
+        eprintln!(
+            "{}: Failed to converge after {} iterations in `{}`:---\n{}\n---",
+            "debug error".red().bold(),
+            MAX_ITERATIONS,
+            fs::relativize_path(path),
+            transformed,
+        );
+    }
+    #[cfg(not(debug_assertions))]
+    {
+        eprintln!(
+            r#"
+{}: Failed to converge after {} iterations.
+
+This indicates a bug in `{}`. If you could open an issue at:
+
+    {}/issues/new?title=%5BInfinite%20loop%5D
+
+...quoting the contents of `{}`, along with the `pyproject.toml` settings and executed command, we'd be very appreciative!
+"#,
+            "error".red().bold(),
+            MAX_ITERATIONS,
+            env!("CARGO_PKG_NAME"),
+            env!("CARGO_PKG_REPOSITORY"),
+            fs::relativize_path(path),
+        );
+    }
+}
+
+#[allow(clippy::print_stderr)]
+fn report_autofix_syntax_error(path: &Path, transformed: &str, error: &ParseError) {
+    #[cfg(debug_assertions)]
+    {
+        eprintln!(
+            "{}: Autofix introduced a syntax error in `{}`:\n\n{}\n\n---\n{}\n---",
```

Comment:
Generating a nice readable error is possible as you can see in, but it's a bit cumbersome to do so at the moment.

https://github.com/charliermarsh/ruff/blob/719395f487018783b20c82198f61dc0f2a229bc6/crates/ruff/src/test.rs#L116-L119

Reply:
@MichaReiser Maybe once that code is in `main` :)